crates/bevy_sprite/src/texture_sheet.rs

```diff
@@ -0,0 +1,39 @@
+use crate::{Rect, TextureAtlas};
+use bevy_asset::{Assets, Handle};
+use bevy_ecs::component::Component;
+use bevy_reflect::Reflect;
+
+/// Component used to draw a specific section of a texture.
+///
+/// It stores a handle to [`TextureAtlas`] and the index of the current section of the atlas.
+/// The texture atlas contains various *sections* or *cuts* of a given texture, allowing users to have a single
+/// image file for either sprite animation or global mapping.
+/// You can change the [`texture_index`](Self::texture_index) of the sheet to animate the sprite or to pick a *section* of the texture.
+///
+/// Check the following examples for usage:
+/// - [`animated sprite sheet example`](https://github.com/bevyengine/bevy/blob/main/examples/2d/sprite_sheet.rs)
+/// - [`texture atlas example`](https://github.com/bevyengine/bevy/blob/main/examples/2d/texture_atlas.rs)
+#[derive(Component, Default, Debug, Clone, Reflect)]
+pub struct TextureSheet {
```

Comment:
I don't like the name `TextureSheet`. It seems like a combination of `SpriteSheet` and `TextureAtlas` and sounds like it should be something conceptually the same - a sheet of textures, like a sprite sheet or texture atlas. Instead, it contains information about the layout of some sub-textures within a texture atlas texture, and an index that refers to one of those such sub-textures. So it's more like a `TextureAtlasTexture` or `TextureAtlasItem` or something like that.

For that matter, now that `TextureAtlas` does not contain the handle to the texture anymore, it is more of a `TextureAtlasLayout`.

That would then result in a `Handle<Image>` + `Handle<TextureAtlasLayout>` define a texture atlas, and a `TextureAtlasItem` or `TextureAtlasIndex` or something just selects the sub-texture from within the `Image`, according to the `TextureAtlasLayout`. What do you think about this naming setup? I'm not sure it makes sense to tie the index and the atlas layout together, if you're going for splitting them up.

Reply:
That was what my first attempt looked like, but I considered that since there is no use case of a `Handle<TextureAtlasLayout>` without an index I figured it would be easier to merge them in a single additive component.

Overall I agree that the naming is not great and your suggestions make more sense ! For the component split I don't have a strong opinion, both versions suit me, so I'll do whatever is preferred

Reply:
ping @alice-i-cecile , what do you think ?

Reply:
I'm on board with Rob's suggestion here.

Reply:
When I did a similar separation between the layout and the index @mockersf  said in this [comment](https://github.com/bevyengine/bevy/pull/5070#discussion_r903560834):
> It doesn't make sense to have a `TextureIndex` without a `TextureAtlas`

Which is why I preferred to put them in a single component. I think he has a good point

Reply:
My apologies, I'd glossed over this a bit. I think a single component makes sesnse here, but I agree with the complaint about the name clarity.

Reply:
But then the layout has to be duplicated for every entity using that texture atlas texture. There is practically/usually a 1:1 mapping between a `Handle<Image>` and a `Handle<TextureAtlasLayout>` using my definitions. And then there are many indices into it for individual sprites, tiles, animation frames, etc. Why duplicate the layout for every reference to those? @mockersf ?

We could have a bundle to keep the usage clear?:
```rust
#[derive(Component)]
struct TextureAtlasLayout {
...
}

#[derive(Component)]
struct TextureAtlasIndex(usize);

/// A bundle defining a subtexture within a texture atlas
struct TextureAtlasTexture {
    image: Handle<Image>,
    layout: Handle<TextureAtlasLayout>,
    index: TextureAtlasIndex,
}
```

Reply:
#1481 would help us enforce this usage / point users in the right direction.

Reply:
So what do I do ?

Reply:
(Deferring to Francois and Rob here)

Reply:
Personally, I'm with rob on this one. It makes more sense as separate components since they could be used separately. Using the same spritesheet for multiple entities is very common. I agree that it introduces an error state that isn't possible with the merged approach, but it seems like a user would have to almost willingly do it to be in that state.

A TextureAtlasBundle with enough documentation and examples should be more than enough to indicate the relationship between them.

Reply:
I was waiting for a response from @mockersf .